### ISSUE TITLE
Improving translation quality of DynamicConv model

### ISSUE THREAD

**Issue by ManiaCello** [pytorch/fairseq#585]
Hello! I'm training DynamicConv model (lightconv_wmt_en_de_big) with GLU for EN-RU language pair. After 2 epochs I have best validation perplexity 3.69 and BLEU 19.1 for newstest2017. Earlier I trained transformer base (6 layers, 8 heads, 512 dimension size) model (OpenNMT implementation) on the same dataset and after 2 epochs validation perplexity was 7.07 and BLEU 24. 
I tried to adjust beam and lenpen options for DynamicConv and 19.1 BLEU - the best I could get. How could I improve BLEU for this model?
I'm using following command for training:

`nohup python $NMT_DIR/fairseq/train.py $NMT_DIR/ds/bin \
     --distributed-world-size 1 \
     --device-id 0 \
     --arch lightconv_wmt_en_de_big \
     --dropout 0.3 \
     --attention-dropout 0.1 \
     --weight-dropout 0.1 \
     --encoder-glu 1 \
     --decoder-glu 1 \
     --max-epoch 2 \
     --max-tokens 1700 \
     --update-freq 1 \
     --optimizer adam \
     --lr 0.001 \
     --adam-betas '(0.9, 0.998)' \
     --clip-norm 0.0 \
     --lr-scheduler inverse_sqrt \
     --weight-decay 0.0 \
     --warmup-init-lr 1e-7 \
     --warmup-updates 60000 \
     --min-lr 1e-09 \
     --lr-shrink 1 \
     --criterion label_smoothed_cross_entropy \
     --label-smoothing 0.1`

**Comment by jiezhangGt** [comment]
Hello, Maybe you could try to train your model without GLU.

**Comment by huihuifan** [comment]
How much data is available for EN-RU? It's possible with GLU you should increase dropout. Also, I recommend training for more epochs for the model to converge.

**Comment by felixgwu** [comment]
It seems that the model gets much lower validation perplexity, but the BLEU score doesn't improve. 
It's less likely that the model is overfitting. 
In contrast, it might be that there is something weird happened during decoding or evaluation.
Is there some difference between Fairseq and OpenNMT in terms of how the BLEU score is calculated?
For instance, the tokenization methods may be different.
It might be better to see if you can get roughly the same BLEU score on EN-RU pair with the Transformer in Fairseq using the same set of hyperparameters (6 layers, 8 heads, 512 dimension size) first.

**Comment by ManiaCello** [comment]
I combined several corpora and got training set with 6.1 million sentences. For both models I used the same tokenization approach: joint BPE with 30000 merge operations (as it described for EN-RU pair in [Sennrich et al.](http://www.aclweb.org/anthology/P16-1162)). The BLEU was calculated with sacrebleu tool for detokenized sentences.
I'm not sure if dropout is really the reason of low performance: as @felixgwu mentioned, there's no overfitting, because validation loss is pretty low and it keeps descending; besides the dropout is quite big already...
Unfortunately, now I temporarily have no GPU, but I'll try to train the Fairseq's Transformer base model implementation when I'll have one (in 4-5 days, I hope).
